src/components/header/renderer.tsx

```diff
@@ -153,10 +153,19 @@ class Header extends React.Component<Props, State> {
 
     const gistUrl = new URL(url, 'https://gist.github.com');
     const [username, gistId] = gistUrl.pathname.split('/').slice(1);
-
+    const gistCommits = await fetch(`https://api.github.com/gists/${gistId}/commits`).then(r => {
+      if (r.ok) {
```

Comment:
Please refactor. Set the state first outside the conditional.

Reply:
Will do.

Reply:
Lol, I mean, set the state to r.ok